Assets/HoloToolkit/Input/Scripts/Focus/FocusDetails.cs

```diff
@@ -9,6 +10,7 @@ namespace HoloToolkit.Unity.InputModule
     /// FocusDetails struct contains information about which game object has the focus currently.
     /// Also contains information about the normal of that point.
     /// </summary>
+    [Serializable]
     public struct FocusDetails
```

Comment:
While we're at it, this should get moved into it's own file as well.

Reply:
It already is...I also disagree with putting structs/enums/classes/etc into their own file 100% of the time. Sometimes it makes sense, but sometimes you lose context when it's isolated.

Reply:
I disagree that the context is lost.  It def makes it much more readable, which is important in open source projects.

I also find it annoying when people warp a public class in other public classes that don't have any dependencies to the class they're part of.

Reply:
@StephenHodgson, I believe you meant **wrap** a public class in other public classes.  I agree and disagree.  I think if you class looks like a turducken (https://en.wikipedia.org/wiki/Turducken) you're doing something wrong but if your class is really just a data type with constructors that belongs in the other classes namespace it's a perfectly acceptable model.  That's actually a very standard model in game development and all 5-6 engines I've shipped games in.

Reply:
Haha yeah. I'm really taking about exactly that 😆

I just like having the shorter references instead of 5-10 references.
`FirstClass.SecondClass.ThirdClass.FouthClass.YourActualRefetence`

I find anything more than two to be completely unnecessary.

Reply:
I agree that readability is important if it enhances comprehension, but it doesn't always do that. Sometimes a *visible* but hard-to-read relationship can be easier to comprehend than a well-formatted *hidden* relationship.

I'll c&v my earlier point about enums since it applies to this situation as well: Breaking them up into separate files destroys information about their purpose while adding complexity. Devs have to work backwards from variables one-by-one, forensically reconstructing relationships that should be clear at a moment's glance.

It'll always be case-by-case but that's my 2c, anyway.

Also turducken is my new favorite programming analogy.